docs(next.clayui.com): Adds missing migration guides (#2629)
liferay/clay: next.clayui.com/content/docs/migration/migrate-the-clay-components-from-v2-to-v3.md

-   `destroyOnHide` was removed
-   `style` was renamed to `displayType`
-   `message` was removed in favor of using `children`

> -   Use `stripe` value on `variant` property of `ClayAlert` to get the same result of `ClayStripe` component.

Reviewer: ```suggestion
-   `stripe` value on `variant` property of `ClayAlert` will get you the same result of v2's `ClayStripe` component.
```

## ClayButton

> -   Removed icon dependency within `ClayButton` itself. However, was created a high-level component called `ButtonWithIcon` that receives a symbol and creates a button monospaced with a icon.

Reviewer: Since we have ClayButtonWithIcon and thus including the dependency again, we can just remove this line entirely.

## ClayCard

> ClayCard v3 is offered in Low-Level API: decomposed to ClayCard.AspectRatio, ClayCard.Body,ClayCard.Description, ClayCard.Caption and ClayCard.Header;

Reviewer: ```suggestion
ClayCard v3 is offered in composable components such as `ClayCard.AspectRatio`, `ClayCard.Body`, `ClayCard.Description`, `ClayCard.Caption`, and `ClayCard.Header`.
```

See `ClayDropDown` in our storybook for more compositions and examples of use.

> Also, it is offered as a High-Level component called [DropDownWithItems](/docs/components/drop-down.html#claydropdownwithitems).

Reviewer: We go back and forth between capitalizing and not capitalizing "high-level", I think it'd be best to not capitalize and use lower case for all instances.
